### ISSUE TITLE
Index other post date field (post_date_gmt, post_modified, post_modified_gmt) components

### ISSUE THREAD

**Issue by nickdaugherty** [10up/ElasticPress#1774]
**Is your enhancement related to a problem? Please describe.**

Currently, ElasticPress indexes the "components" (like year/month/day/etc) of the `post_date` field, but not the components of the other core date fields `post_date_gmt`, `post_modified`, `post_modified_gmt`. This means that any query attempting to use these fields in the same way as `post_date` will not behave as expected.

**Describe the solution you'd like**

Out of the box, EP indexes all core post date fields in the same way (broken down into components).

**Designs**

This should be pretty easy, mostly just adding the right `Indexable::prepare_date_terms()` calls and the mapping changes.

**Additional context**

Related to #1746 (which is for dates in meta fields)

**Comment by tlovett1** [comment]
@nickdaugherty can you open a PR back to this repo?

**Comment by nickdaugherty** [comment]
Yep that's the plan (just making sure it's stable on our fork first).
